src/loaders/svg/tvgSvgLoader.cpp

```diff
@@ -1913,13 +1913,13 @@ static SvgStyleGradient* _createRadialGradient(SvgLoaderData* loader, const char
         return nullptr;
     }
     /**
-    * Default values of gradient
+    * Default values of gradient transformed into global percentage
     */
-    grad->radial->cx = 0.5;
-    grad->radial->cy = 0.5;
-    grad->radial->fx = 0.5;
-    grad->radial->fy = 0.5;
-    grad->radial->r = 0.5;
+    grad->radial->cx = 0.5f / loader->svgParse->global.w;
+    grad->radial->cy = 0.5f / loader->svgParse->global.h;
+    grad->radial->fx = 0.5f / loader->svgParse->global.w;
+    grad->radial->fy = 0.5f / loader->svgParse->global.h;
```

Comment:
If the viewbox and width and height are set to 0 in the svg file, global.w and h can be 0.
It's a wrong svg file but we need to make sure SvgLoader doesn't crash.
Even if you make a patch for global.w and h later, I think we need the logic to check zero now.

how about this?
```cpp
auto w = loader->svgParse->global.w  != 0 ? loader->svgParse->global.w : 1;
...
```

Reply:
@JSUYA please give me some time to think about the best solution

Reply:
Return immediately in the beginning of validation header. (if size is invalid)

Reply:
it's just a temporary solution. For now I made sure, that the global variables are not zeros.
I don't want to add the check as @JSUYA suggested, since it should be done earlier, when the svg tag appears. The coordinate system is related to the attributes, that are present in this tag. And the global variables representing this coordinate system.

Reply:
@mgrudzinska  I mentioned we can handle it in the header validation (of Svg Loader), It's an exceptional handling for safer program. Anyway no one reject this patch, and that topic is another one.